### ISSUE TITLE
Broken Multicluster Installation Task

### ISSUE THREAD

**Issue by nschhina** [istio/istio#26017]
Creating this issue for 1.7 bug handling for multicluster here.
The new user guide is still in review so it's a given that there's some bugs in documenting this task.
[User Task] https://github.com/istio/istio.io/pull/7787
[Preview]  https://deploy-preview-7787--preliminary-istio.netlify.app/latest/docs/setup/install/multicluster/single-network/

I've run into following feature issues:

During initial configuration as well as while adding Remote/Primary cluster
Generate the root CA and the intermediate CA for the cluster with the following command:
`make -f ${ISTIO}/tools/certs/Makefile ${CLUSTER_1}-cacerts-k8s`

This command was earlier pointing to a Makefile in `install/certs`, but in 1.6 onwards the Makefile was changed and moved to `tools/certs` earlier the make command used to generate certs was `${NAME}-certs`. I'm not sure what the new Make command is. Generating certs with Makefile from Istio 1.5 can be used to get by this issue.

Second and a more critical issue: 
Remote Cluster does not work correctly with Primary Cluster's `istiod`. 
Steps to reproduce:
1. Deploy initial cluster
2. Try adding a primary cluster
3. Do the steps in verify your deployment.
4. Try deploying a sidecar injected helloworld app in remote cluster following the `verify your deployment` guide.

istio-proxy container for helloworld app in remote cluster never comes live. 

Logs from istio-proxy in helloworld pod:
`Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 0 successful, 0 rejected; lds updates: 0 successful, 0 rejected`

Logs from istiod in remote:
```
2020-07-30T21:47:07.878563Z	info	ads	Push debounce stable[4] 5: 100.148341ms since last change, 172.94924ms since last push, full=true
2020-07-30T21:47:07.879276Z	info	ads	XDS: Pushing:2020-07-30T21:47:07Z/4 Services:9 ConnectedEndpoints:0
2020-07-30T21:47:08.479441Z	info	ads	Push debounce stable[5] 1: 100.190578ms since last change, 100.190344ms since last push, full=false
2020-07-30T21:47:08.479884Z	info	ads	XDS:EDSInc Pushing:2020-07-30T21:47:07Z/4 Services:map[istio-ingressgateway.istio-system.svc.cluster.local:{}] ConnectedEndpoints:0
2020-07-30T21:47:08.973286Z	info	ads	Push debounce stable[6] 1: 100.184855ms since last change, 100.184649ms since last push, full=false
2020-07-30T21:47:08.973680Z	info	ads	XDS:EDSInc Pushing:2020-07-30T21:47:07Z/4 Services:map[istio-egressgateway.istio-system.svc.cluster.local:{}] ConnectedEndpoints:0
2020-07-30T21:47:09.492411Z	info	ads	Full push, new service istio-egressgateway.istio-system.svc.cluster.local
2020-07-30T21:47:09.592716Z	info	ads	Push debounce stable[7] 1: 100.167655ms since last change, 100.167435ms since last push, full=true
2020-07-30T21:47:09.593216Z	info	ads	XDS: Pushing:2020-07-30T21:47:09Z/5 Services:9 ConnectedEndpoints:0
2020-07-30T21:47:10.772946Z	info	ads	Full push, new service istio-ingressgateway.istio-system.svc.cluster.local
2020-07-30T21:47:10.873281Z	info	ads	Push debounce stable[8] 1: 100.187272ms since last change, 100.187088ms since last push, full=true
2020-07-30T21:47:10.873898Z	info	ads	XDS: Pushing:2020-07-30T21:47:10Z/6 Services:9 ConnectedEndpoints:0
2020-07-30T21:47:12.048209Z	info	ads	Push Status: {}
2020-07-30T21:47:45.907413Z	info	ads	Push debounce stable[9] 1: 100.220431ms since last change, 100.220197ms since last push, full=true
2020-07-30T21:47:45.907990Z	info	ads	XDS: Pushing:2020-07-30T21:47:45Z/7 Services:9 ConnectedEndpoints:0
2020-07-30T21:47:52.048079Z	info	ads	Push Status: {}
2020-07-30T21:48:12.333042Z	info	Handle EDS endpoint: skip updating, service helloworld/sample has not been populated
2020-07-30T21:48:12.417848Z	info	ads	Push debounce stable[10] 1: 100.268112ms since last change, 100.267737ms since last push, full=true
2020-07-30T21:48:12.418618Z	info	ads	XDS: Pushing:2020-07-30T21:48:12Z/8 Services:8 ConnectedEndpoints:0
2020-07-30T21:48:22.048047Z	info	ads	Push Status: {}
2020-07-30T21:48:33.355048Z	info	ads	Push debounce stable[11] 2: 100.201299ms since last change, 111.044048ms since last push, full=true
2020-07-30T21:48:33.355550Z	info	ads	XDS: Pushing:2020-07-30T21:48:33Z/9 Services:9 ConnectedEndpoints:0
2020-07-30T21:48:35.104268Z	info	AdmissionReview for Kind=/v1, Kind=Pod Namespace=sample Name= (helloworld-v2-776f74c475-***** (actual name not yet known)) UID=88f6f23d-a956-4336-8442-ad9739938694 Rfc6902PatchOperation=CREATE UserInfo={system:serviceaccount:kube-system:replicaset-controller e2a30f71-960c-4de1-b9aa-2d91c8a9619e [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] map[]}
2020-07-30T21:48:36.854586Z	info	ads	Push debounce stable[12] 1: 100.207151ms since last change, 100.206952ms since last push, full=false
2020-07-30T21:48:36.854688Z	info	ads	XDS:EDSInc Pushing:2020-07-30T21:48:33Z/9 Services:map[helloworld.sample.svc.cluster.local:{}] ConnectedEndpoints:0
2020-07-30T21:48:42.048110Z	info	ads	Push Status: {}
```

Logs from istiod in primary cluster:

```
2020-07-30T21:48:39.844186Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Unauthorized
2020-07-30T21:48:56.891548Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Service: Unauthorized
2020-07-30T21:48:57.106406Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Unauthorized
2020-07-30T21:49:20.289474Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:49:25.948554Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:49:29.272108Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:49:36.932513Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Service: Unauthorized
2020-07-30T21:49:39.340650Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Unauthorized
2020-07-30T21:49:43.261733Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:49:46.406920Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:49:48.011703Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:49:56.307653Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Node: Unauthorized
2020-07-30T21:50:17.567340Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:50:19.014094Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Unauthorized
2020-07-30T21:50:19.089760Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:50:19.634474Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:50:33.335838Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:50:33.830132Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Service: Unauthorized
2020-07-30T21:50:45.651085Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Node: Unauthorized
2020-07-30T21:50:53.677884Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:51:05.080662Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Unauthorized
2020-07-30T21:51:07.219864Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to watch *v1.Pod: the server has asked for the client to provide credentials (get pods)
2020-07-30T21:51:08.375341Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Pod: Unauthorized
2020-07-30T21:51:10.986099Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Pod: Unauthorized
2020-07-30T21:51:11.003818Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Service: Unauthorized
2020-07-30T21:51:15.953514Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Pod: Unauthorized
2020-07-30T21:51:19.596177Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:51:25.925143Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:51:27.028485Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Pod: Unauthorized
2020-07-30T21:51:34.501938Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Node: Unauthorized
2020-07-30T21:51:43.527114Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Unauthorized
2020-07-30T21:51:50.467311Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Pod: Unauthorized
2020-07-30T21:51:51.326362Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:51:59.750573Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Service: Unauthorized
2020-07-30T21:52:00.372104Z	warn	serverca	Authentication failed for 10.60.8.2:51854: Authenticator ClientCertAuthenticator at index 0 got error: no verified chain is found. Authenticator KubeJWTAuthenticator at index 1 got error: failed to validate the JWT: tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope. Authenticator ClientCertAuthenticator at index 2 got error: no verified chain is found. 
2020-07-30T21:52:04.525383Z	error	k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Node: Unauthorized
```

**Comment by stevenctl** [comment]
1. Looks like the k8s-root-cert and k8s-root-key files are empty for the new certs make target
2. We need to fix indentation errors for meshNetworks on the [initial configuration](https://deploy-preview-7787--preliminary-istio.netlify.app/latest/docs/setup/install/multicluster/single-network/initial-configuration/) page
3. The comment about `Always use ‘Kubernetes' as the registry name` is inaccurate AFAIK. It should match `clusterName`.

**Comment by stevenctl** [comment]
Also, the docs use `manifest apply` and the error message isn't very informative. Should be install. 

```
 istioctl --context=${CTX_1} manifest apply -f ${WORK_DIR}/${CLUSTER_1}/install.yaml               ✘ 130
Error: unknown shorthand flag: 'f' in -f
```

**Comment by stevenctl** [comment]
This would be a more-accurate installation for the main cluster: 

```
cat <<EOF> ${WORK_DIR}/${CLUSTER_1}/install.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      # Expose the control plane through istio-ingressgateway.
      meshExpansion:
        enabled: true
      meshID: ${MESH}
      multiCluster:
        clusterName: ${CLUSTER_1}
      network: ${NETWORK_1}
      meshNetworks:
        ${NETWORK_1}:
        endpoints:
          # fromRegistry should match the clusterName used above
          # If you plan on having more than one cluster on the same network,
          # you should have a fromRegistry entry for each cluster.
          - fromRegistry: ${CLUSTER_1}
        gateways:
          - registry_service_name: istio-ingressgateway.istio-system.svc.cluster.local
        port: 443
EOF
```

**Comment by stevenctl** [comment]
And there is a missing setting int he remote installation settings. The remote cluster should use the local istiod for CA. The following is the proper template

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: ${MESH}
      multiCluster:
        clusterName: ${CLUSTER_1}
      network: ${NETWORK_1}
      # Access the control plane discovery server via the ingress
      remotePilotAddress: ${DISCOVERY_ADDRESS}
      # Use the local istiod for CA
      caAddress: istiod.istio-system.svc:15012
```

**Comment by stevenctl** [comment]
cc @rcaballeromx ^

**Comment by nschhina** [comment]
I retried with the suggestions from @stevenctl and I'm able to sucessfully establish cross-cluster communication now between `primary-remote`

**Comment by sdake** [comment]
cc /@sdake

**Comment by therealmitchconnors** [comment]
+1 for the problems with makefiles. similarly affecting vm docs. https://github.com/istio/istio.io/issues/7855

**Comment by sdake** [comment]
This instruction does not work if dns is used in the ingress host:

```
export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
```
from the page: https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/#determining-the-ingress-ip-and-ports

**Comment by sdake** [comment]
I don't find the warning sufficient in that document. Atleast there should be a warning before the problem that can occur, followed by corrective steps.

**Comment by sdake** [comment]
lots of warnings in 1.6.8 on step:
```
curl -LO https://storage.googleapis.com/istio-release/releases/1.6.8/deb/istio-sidecar.deb
sudo dpkg -i istio-sidecar.deb

```

```
root@sdake-vm:~# sudo dpkg -i istio-sidecar.deb
Selecting previously unselected package istio-sidecar.
(Reading database ... 90451 files and directories currently installed.)
Preparing to unpack istio-sidecar.deb ...
Unpacking istio-sidecar (1.6.8) ...
Setting up istio-sidecar (1.6.8) ...
Adding group `istio-proxy' (GID 115) ...
Done.
Warning: The home dir /var/lib/istio you specified already exists.
Adding system user `istio-proxy' (UID 111) ...
Adding new user `istio-proxy' (UID 111) with group `istio-proxy' ...
The home directory `/var/lib/istio' already exists.  Not copying from `/etc/skel'.
adduser: Warning: The home directory `/var/lib/istio' does not belong to the user you are currently creating.
```

**Comment by stevenctl** [comment]
@sdake are you following the guide linked in the description? I don't think that's what we're going with for 1.7

**Comment by sdake** [comment]
@stevenctl the decision was made in the workgroup lead meeting today to do the following:
1. stabilize the existing virtual machine integration documentation for the 1.7 release, as the branch Is tomorrow am.
1. After the existing 1.6 docs are stabilized, then think about adding the next-gen security model for 1.7.

I am not confident, after having seen 200+ comments on the original review of this document, that a new workflow can be merged in one to two days. So we are taking a path of least risk. Feel free to continue defining a new document, however, we need working code in 1.7 with no regressions.

**Comment by sdake** [comment]
Ubuntu in my environment (IKS) presents the following resolv.conf which does not reference /etc/hosts.

```
root@sdake-vm:/var/log/istio# cat /etc/resolv.conf
# This file is managed by man:systemd-resolved(8). Do not edit.
#
# This is a dynamic resolv.conf file for connecting local clients to the
# internal DNS stub resolver of systemd-resolved. This file lists all
# configured search domains.
#
# Run "systemd-resolve --status" to see details about the uplink DNS servers
# currently in use.
#
# Third party programs must not access this file directly, but only through the
# symlink at /etc/resolv.conf. To manage man:resolv.conf(5) in a different way,
# replace this symlink by a static file or a different symlink.
#
# See man:systemd-resolved.service(8) for details about the supported modes of
# operation for /etc/resolv.conf.

nameserver 127.0.0.53
options edns0
```

This will need to be explained, as Ubuntu is the default..

**Comment by howardjohn** [comment]
I am fairly sure /etc/hosts is still used - I just played around with DNS a lot in Ubuntu 18 on GCE. The resolv.conf is identical but it will read /etc/hosts

**Comment by sdake** [comment]
this looks good:

```
020-08-12T03:06:34.217324Z     info    sds     resource:ROOTCA new connection
2020-08-12T03:06:34.217399Z     info    sds     Skipping waiting for ingress gateway secret
2020-08-12T03:06:34.217663Z     info    cache   adding watcher for file ./etc/certs/root-cert.pem
2020-08-12T03:06:34.217714Z     info    cache   GenerateSecret from file ROOTCA
2020-08-12T03:06:34.218507Z     info    sds     resource:ROOTCA pushed root cert to proxy
2020-08-12T03:06:48.904373Z     info    Subchannel Connectivity change to CONNECTING
2020-08-12T03:06:48.904444Z     info    Subchannel picks a new address "istiod.istio-system.svc:15012" to connect
2020-08-12T03:06:48.904483Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc000c60830, {CONNECTING <nil>}
2020-08-12T03:06:48.904508Z     info    Channel Connectivity change to CONNECTING
2020-08-12T03:06:48.937576Z     info    Subchannel Connectivity change to READY
2020-08-12T03:06:48.937959Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc000c60830, {READY <nil>}
2020-08-12T03:06:48.937998Z     info    Channel Connectivity change to READY
```

**Comment by sdake** [comment]
As my hosts are DNS, a DNS name followed by a DNS name in /etc/hosts is not sufficient. Instead, the first DNS name should be resolved:
e.g. my hosts looks like this:

```
root@sdake-vm:/var/log/istio# cat /etc/hosts
127.0.0.1       localhost

# The following lines are desirable for IPv6 capable hosts
::1     ip6-localhost   ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
ff02::3 ip6-allhosts
127.0.1.1       sdake-vm        sdake-vm

52.117.8.137 istiod.istio-system.svc
```

while the `hosts-addendum` file looks like this:

```
root@sdake-vm:~# cat hosts-addendum 
7d7cb96a-us-south.lb.appdomain.cloud istiod.istio-system.svc
```

**Comment by sdake** [comment]
Oh, my apologies. I am not addressing the multicluster docs @stevenctl . I am addressing only virtual machine integration. I'll take this discussion to https://github.com/istio/istio.io/issues/7855.

**Comment by stevenctl** [comment]
Closing this, I think we resolved it for the 1.7 release.
